product_tax_multicompany_default/tests/test_product_tax_multicompany.py

```diff
@@ -101,27 +120,35 @@ def test_multicompany_default_tax(self):
 
     @users("user_1")
     def test_not_default_tax_if_set(self):
-        product = self.env["product.product"].create(
-            {
-                "name": "Test Product",
-                "taxes_id": [(6, 0, self.tax_20_cc1.ids)],
-                "supplier_taxes_id": [(6, 0, self.tax_20_sc1.ids)],
-                "company_id": False,
-            }
+        product = (
+            self.env["product.product"]
+            .sudo()
```

Comment:
Why do you need the sudo? It shouldn't be needed.

Reply:
@pedrobaeza In the PR description, I justified these changes. Basically, the `account.group_account_manager` group does not have permission to create products.

I also added comments in the code to clarify and removed some unnecessary `sudo()` calls after create. Let me know if you have any other comments. Thank you.